connector/web3/web3.go

```diff
@@ -37,32 +47,73 @@ func (c *web3Connector) Verify(address, msg, signedMsg string) (identity connect
 		return identity, fmt.Errorf("could not decode hex string of signed nonce: %v", err)
 	}
 
+	msgHash := signHash([]byte(msg))
+
 	// This is the v parameter in the signature. Per the yellow paper, 27 means even and 28
 	// means odd.
 	if signb[64] == 27 || signb[64] == 28 {
 		signb[64] -= 27
 	} else if signb[64] != 0 && signb[64] != 1 {
 		// We allow 0 or 1 because some non-conformant devices like Ledger use it.
-		return identity, fmt.Errorf("byte at index 64 of signed message should be 27 or 28: %s", signedMsg)
+		return c.VerifyERC1271Signature(addrb, msgHash, signedMsg)
 	}
 
-	pubKey, err := crypto.SigToPub(signHash([]byte(msg)), signb)
+	pubKey, err := crypto.SigToPub(msgHash, signb)
 	if err != nil {
-		return identity, fmt.Errorf("failed to recover public key from signed message: %v", err)
+		return c.VerifyERC1271Signature(addrb, msgHash, signedMsg)
 	}
 
 	recoveredAddr := crypto.PubkeyToAddress(*pubKey)
 	// These are byte arrays, so this is okay to do.
 	if recoveredAddr != addrb {
-		return identity, fmt.Errorf("given address and address recovered from signed nonce do not match")
+		return c.VerifyERC1271Signature(addrb, msgHash, signedMsg)
```

Comment:
Do we have a test that checks that we can get down here?

Reply:
Again, I think it would be good to structure this, in the end, as 

```
if EOA check succeeds:
  return good
if contract check succeeds:
  return good
return bad
```

Something like that. These multiple exit points (and it was bad before you got here!) are asking for trouble.

Reply:
Yes there is [https://github.com/DIMO-Network/dex/pull/178/files/abf2a2875271de55951617e5c1d62657d3ddfa35#diff-a6d520ac66b4f80b4c75d315f932f31a0d86754df4de1e94e8ea985394aeada2R132](https://github.com/DIMO-Network/dex/pull/178/files/abf2a2875271de55951617e5c1d62657d3ddfa35#diff-a6d520ac66b4f80b4c75d315f932f31a0d86754df4de1e94e8ea985394aeada2R132)

Reply:
> Do we have a test that checks that we can get down here?

Yes there's a test for that [https://github.com/DIMO-Network/dex/pull/178/files/abf2a2875271de55951617e5c1d62657d3ddfa35#diff-a6d520ac66b4f80b4c75d315f932f31a0d86754df4de1e94e8ea985394aeada2R132](https://github.com/DIMO-Network/dex/pull/178/files/abf2a2875271de55951617e5c1d62657d3ddfa35#diff-a6d520ac66b4f80b4c75d315f932f31a0d86754df4de1e94e8ea985394aeada2R132)

Reply:
The reason I'm concerned is that we originally didn't have this branch, and then we changed it but I didn't see any new tests, so it feels like something must be off. Maybe the tests changed in subtle ways, I could be wrong!

I still think we should simplify the flow. I'll go over this more carefully tonight and we can talk maybe... tomorrow? Today's packed.

Reply:
I think these tests are the good ones https://github.com/DIMO-Network/dex/pull/178/files#diff-a6d520ac66b4f80b4c75d315f932f31a0d86754df4de1e94e8ea985394aeada2R234-R264

Reply:
> Again, I think it would be good to structure this, in the end, as
> 
> ```
> if EOA check succeeds:
>   return good
> if contract check succeeds:
>   return good
> return bad
> ```
> 
> Something like that. These multiple exit points (and it was bad before you got here!) are asking for trouble.

How do you feel about this?